src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -130,6 +130,18 @@ public void AddProject(PackageSpec projectSpec)
             _projects.Add(projectUniqueName, projectSpec);
         }
 
+        /// <summary>
+        /// Add all restorable projects to the restore list.
```

Comment:
Actually if project is in restore list already it won't be added. So maybe Ensure?